### ISSUE TITLE
SearchableList.seperated can't filter correctly.

### ISSUE THREAD

**Issue by crimson-gao** [koukibadr/Searchable-Listview#54]
There are a few problems I have found.
The code comes from offical [examples](https://github.com/koukibadr/Searchable-Listview/blob/main/example/lib/main.dart). These lines in the picture are all what I have modified.
from
```dart
child: SearchableList<Actor>(
                style: const TextStyle(fontSize: 25),
```
to
```dart
child: SearchableList<Actor>.seperated(
                seperatorBuilder: (context, i) => const Divider(),
                // initialList: actors,
                style: const TextStyle(fontSize: 25),
```
Here comes the first problem. When I pass **no** `initalList` to `SearchableList<Actor>.seperated`, it would throw a runtime exception around these codes. It seems that seperated list requires parameter `initialList`, even if parameter `asyncListCallback` is set. But the document only declares that `'either initialList or asyncListCallback must be provided'`.
```dart
Widget _renderSeperatedListView() {
    return ListView.separated(
      controller: scrollController,
      scrollDirection: widget.scrollDirection,
      itemCount: widget.initialList!.length,   // here
      itemBuilder: (context, index) => ListItem<T>(
        builder: widget.builder,
        item: widget.initialList![index],
        onItemSelected: widget.onItemSelected,
      ),
      separatorBuilder: widget.seperatorBuilder!,
    );
  }
```
Then I add the `initialList`, so now my code is as follows.
```dart
child: SearchableList<Actor>.seperated(
                seperatorBuilder: (context, i) => const Divider(),
                initialList: actors,
                style: const TextStyle(fontSize: 25),
```

The code runs well, but when as soon as I uses the search bar to filter list items, I found that the filter does not work. 
If there are still any item which meets the filter condition, then the list keeps the original state.

![image](https://user-images.githubusercontent.com/39024757/231958034-9b0595a7-cddc-40d1-a143-36bfebd27446.png)
 
If every items fail to meet the condition, then suddenly the list becomes empty as expected.

![image](https://user-images.githubusercontent.com/39024757/231958463-1da26d92-6343-4861-a757-e11500b84e7d.png)

Here are my environments.
```
Flutter 3.7.11 • channel stable • https://github.com/flutter/flutter.git
Framework • revision f72efea43c (2 days ago) • 2023-04-11 11:57:21 -0700
Engine • revision 1a65d409c7
Tools • Dart 2.19.6 • DevTools 2.20.1
```
device: Pixel_XL_API_33 (android_x64_emulator)

Thanks a lot.

**Comment by crimson-gao** [comment]
fix in #55

**Comment by crimson-gao** [comment]
The problem is function `_renderSeperatedListView` uses initalList instead of the filterResult, even if `asyncListFilter` and `asyncListCallback` is set.
Thus, the filter makes no difference here. But if there are no matching results, it renders an empty list.

```dart
Widget _renderListView({
    required List<T> list,
  }) {
    if (list.isEmpty) {
      return widget.emptyWidget;
    } else {
      return Expanded(
        child: widget.onRefresh != null
            ? RefreshIndicator(
                triggerMode: RefreshIndicatorTriggerMode.onEdge,
                onRefresh: widget.onRefresh!,
                child: widget.displayDividder
                    ? _renderSeperatedListView(list: list)
                    : ListView.builder(
                        controller: scrollController,
                        scrollDirection: widget.scrollDirection,
                        itemCount: list.length,
                        itemBuilder: (context, index) => ListItem<T>(
                          builder: widget.builder,
                          item: list[index],
```
